src/main/java/com/wanderersoftherift/wotr/item/riftkey/RiftConfig.java

```diff
@@ -20,34 +23,43 @@
 /**
  * Configuration for generating a rift
  *
- * @param tier  The tier of the rift
- * @param theme The theme of the rift
- * @param seed  Optional, the seed for the rift
+ * @param tier      The tier of the rift
+ * @param theme     The theme of the rift
+ * @param objective The objective of the rift
+ * @param seed      Optional, the seed for the rift
  */
-public record RiftConfig(int tier, Optional<Holder<RiftTheme>> theme, Optional<Integer> seed) {
+public record RiftConfig(int tier, Optional<Holder<RiftTheme>> theme, Optional<Holder<ObjectiveType>> objective,
+        Optional<Integer> seed) {
 
     public static final Codec<RiftConfig> CODEC = RecordCodecBuilder
             .create(instance -> instance
                     .group(Codec.INT.fieldOf("tier").forGetter(RiftConfig::tier),
                             RiftTheme.CODEC.optionalFieldOf("theme").forGetter(RiftConfig::theme),
+                            RegistryFixedCodec.create(RegistryEvents.OBJECTIVE_REGISTRY)
```

Comment:
Should probably move this line to something like Objective.CODEC. That would be more consistent with everything else.

Reply:
Done